agent/config/parse_windows_test.go

```diff
@@ -0,0 +1,30 @@
+// +build windows,unit
+
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package config
+
+import (
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestParseGMSACapability(t *testing.T) {
```

Comment:
Adding that test requires mocking syscalls to windows. Are there any interfaces setup for that purpose?

Reply:
You can mock your helper function instead of mocking sys calls.
in agent/config/parse_windows.go
```golang
var (
    isDomainJoinedFunction = isDomainJoined
}
```

in test file
```golang
isDomainJoinedFunction = func (bool, error) {
    // your mock logic goes here
}
```

Reply:
I do not like this pattern at all. 
However, if you think this is strong requirement,  I will use the proposed idea.

Reply:
i don't have strong opinion on the test approach. but i do think you should test the true case. If your planned functional tests will cover the case. That's ok with me.

Reply:
The functional tests will address these cases.